bindings/c/test/apitester/TesterTransactionExecutor.cpp

```diff
@@ -399,6 +402,7 @@ class AsyncTransactionContext : public TransactionContextBase {
 			scheduler->schedule(cbInfo.cont);
 			return;
 		}
+		lock.unlock();
```

Comment:
It is indeed a bit complicated. The mutex controls access to the txState and callbackMap, but the txState determines the availability of certain state and actions. In particular, the callback should be executed only in the IN_PROGRESS state, and execution of callbacks require access to the transaction and the scheduler. So it is important to prevent further state transitions while accessing these objects. After transition to other states ON_ERROR and DONE, we establish an exclusive access to the state of the TransactionContext by one thread at a time, so we access without holding a lock.